tests/testthat/test-shinytest2-tm_g_scatterplot.R

```diff
@@ -0,0 +1,194 @@
+            fixed = FALSE
+          )
+        ),
+        size_by = data_extract_spec(
+          dataname = "ADSL",
+          select = teal.transform::select_spec(
+            label = "Select variable:",
+            choices = teal.transform::variable_choices(data[["ADSL"]], c("AGE", "BMRKR1")),
+            selected = "AGE",
+            multiple = FALSE,
+            fixed = FALSE
+          )
+        ),
+        row_facet = data_extract_spec(
+          dataname = "ADSL",
+          select = teal.transform::select_spec(
+            label = "Select variable:",
+            choices = teal.transform::variable_choices(data[["ADSL"]], c("BMRKR2", "RACE", "REGION1")),
+            selected = NULL,
+            multiple = FALSE,
+            fixed = FALSE
+          )
+        ),
+        col_facet = data_extract_spec(
+          dataname = "ADSL",
+          select = teal.transform::select_spec(
+            label = "Select variable:",
+            choices = teal.transform::variable_choices(data[["ADSL"]], c("BMRKR2", "RACE", "REGION1")),
+            selected = NULL,
+            multiple = FALSE,
+            fixed = FALSE
+          )
+        ),
+        ggplot2_args = teal.widgets::ggplot2_args(
+          labs = list(subtitle = "Plot generated by Scatterplot Module")
+        )
+      )
+    )
+  )
+}
+
+testthat::test_that("e2e - tm_g_scatterplot: module is initialised with the specified defaults", {
+  skip_if_too_deep(5)
+
+  app <- app_driver_tm_g_scatterplot()
+
+  app$expect_no_shiny_error()
+
+  testthat::expect_equal(app$get_active_module_input("x-dataset_ADSL_singleextract-select"), "AGE")
+  testthat::expect_equal(app$get_active_module_input("y-dataset_ADSL_singleextract-select"), "BMRKR1")
+  testthat::expect_false(app$get_active_module_input("log_x"))
+  testthat::expect_false(app$get_active_module_input("log_y"))
+  testthat::expect_null(app$get_active_module_input("color_by-dataset_ADSL_singleextract-select"))
+  testthat::expect_equal(app$get_active_module_input("size_by-dataset_ADSL_singleextract-select"), "AGE")
+  testthat::expect_null(app$get_active_module_input("row_facet-dataset_ADSL_singleextract-select"))
+  testthat::expect_null(app$get_active_module_input("col_facet-dataset_ADSL_singleextract-select"))
+  testthat::expect_equal(app$get_active_module_input("alpha"), 1)
+  testthat::expect_equal(app$get_active_module_input("shape"), "circle")
+  testthat::expect_equal(app$get_active_module_input("color"), "#000000")
+  testthat::expect_equal(app$get_active_module_input("size"), 5)
+  testthat::expect_false(app$get_active_module_input("rotate_xaxis_labels"))
+  testthat::expect_false(app$get_active_module_input("add_density"))
+  testthat::expect_false(app$get_active_module_input("rug_plot"))
+  testthat::expect_false(app$get_active_module_input("show_count"))
+  testthat::expect_null(app$get_active_module_input("smoothing_degree"))
+  testthat::expect_equal(app$get_active_module_input("ggtheme"), "gray")
```

Comment:
or `teal.widgets::ggplot2_args`

Reply:
If it's configurable on the module parameters and encoding I think that it fits well here.

This is the biggest integration test that we can do `{tmg}` + `{widgets}` + `{teal}`. But it could still fit with `{teal.widgets}`